src/codegen/compile_enhanced.ml

```diff
@@ -815,13 +834,12 @@ module E = struct
 
   let object_pool_add (env : t) line (allocation : t -> G.t) : int64 =
     if !(env.object_pool.frozen) then raise (Invalid_argument "Object pool frozen");
-    let index = !(env.object_pool.length) in
-    env.object_pool.objects := (line, allocation) :: !(env.object_pool.objects);
-    env.object_pool.length := index + 1;
+    let index, t = Table.add !(env.object_pool.objects) (line, allocation) in
```

Comment:
For the record: this is adding to the back not the front, but order does not matter here, right?